### DISCUSSION TITLE
How to use buses

### DISCUSSION THREAD

**Discussion by AutumnalBlake** [DigitalLogicSimCommunity/Digital-Logic-Sim-CE#54] [Q&A] [answered]
I'm trying to make an 8-bit register from 8 flip-flops. Can I have an 8-bit bus as the input and split it into the registers?

**Comment by AOx0** [comment, marked as answer]
To change the input type you need to select one from the drop-down menu.
<img width="512" alt="Screenshot 2023-01-05 at 12 46 28 a m" src="https://user-images.githubusercontent.com/50227494/210719187-09cb3b83-b6ea-45e4-9b39-76f69e058618.png">

You can only change the type of a single pin. Pin groups are only simple bits
<img width="330" alt="Screenshot 2023-01-05 at 12 51 53 a m" src="https://user-images.githubusercontent.com/50227494/210719345-cb271e11-28f3-4dba-acd4-6f2072b6be20.png">

Then you would need a decoder to extract the bus info, here I save that as a chip and then test it.
<img width="1680" alt="Screenshot 2023-01-05 at 12 48 01 a m" src="https://user-images.githubusercontent.com/50227494/210719501-b054eef8-e6b4-46ba-944f-69855dd20334.png">
<img width="1680" alt="Screenshot 2023-01-05 at 12 49 55 a m" src="https://user-images.githubusercontent.com/50227494/210719560-bc73ea7b-f118-4151-8869-2e7a6459eeb2.png">

Hope this helps. Let me know.

**Reply by AutumnalBlake** [reply]
Thanks :)

**Reply by AutumnalBlake** [reply]
I made a chip with 8-wide input and output, but they keep reverting back to simple. Is this normal?

**Reply by AOx0** [reply]
When they switch to simple are you unable to connect to/from them with another 8-bit wire? I say this because the type may show Simple as the value in the drop-down menu but actually be 8-bit bus.

**Reply by AutumnalBlake** [reply]
It is letting me connect them

**Reply by AOx0** [reply]
Then you're all fine. It's just a visual bug. You should be able to continue normally. If not, then please let me know.

**Reply by AutumnalBlake** [reply]
Is there any way to use the 32 bit bus, since there's no 32 bit decoder?

**Reply by AOx0** [reply]
![giphy](https://user-images.githubusercontent.com/50227494/210731433-10f649fc-e12f-4ec2-8fd8-a8ffc5b273c2.gif)

Seriously tho. As far as I know, there's no way.
